docs/tutorials/01_introductory/viz_solar_system.py

```diff
@@ -163,137 +163,70 @@ def get_orbital_position(radius, time):
-# Define the ``timer_callback`` function, which controls what events happen
-# at certain times, using the counter. Redefine the position of each planet
-# actor using ``get_orbital_position,`` assigning the x and y values of
-# each planet's position with the newly calculated ones. Append each new
-# planet position to its corresponding track array.
-
-def timer_callback(_obj, _event):
-    cnt = next(counter)
-    showm.render()
-
-    global mercury_track, venus_track, earth_track, mars_track, jupiter_track
-    global saturn_track, uranus_track, neptune_track
+# ``calculate_path`` function is for calculating the path/orbit
+# of every planet.
 
-    update_planet_position(r_mercury, mercury_actor, mercury_track, cnt)
 
-    update_planet_position(r_venus, venus_actor, venus_track, cnt)
+def calculate_path(r_planet, planet_track, cnt):
+    for i in range(cnt):
+        pos_planet = get_orbital_position(r_planet, i)
+        planet_track.append([pos_planet[0], 0, pos_planet[1]])
 
-    update_planet_position(r_earth, earth_actor, earth_track, cnt)
 
-    update_planet_position(r_mars, mars_actor, mars_track, cnt)
-
-    update_planet_position(r_jupiter, jupiter_actor, jupiter_track, cnt)
-
-    pos_saturn = update_planet_position(r_saturn, saturn_actor, saturn_track,
-                                        cnt)
-    saturn_rings_actor.SetPosition(pos_saturn[0], 0, pos_saturn[1])
-
-    update_planet_position(r_uranus, uranus_actor, uranus_track, cnt)
-
-    update_planet_position(r_neptune, neptune_actor, neptune_track, cnt)
+##############################################################################
+# First we are making two lists that will contain radii and track's lists.
+# `planet_tracks` is list of 8 empty list for 8 different planets.
+# and `planet_actors` will contain all the planet actor.
+# and then we are calculating and updating the path/orbit
+# before animation starts.
 
-    if cnt == 999:
-        update_track(positions_mercury, mercury_track, mercury_orbit_actor)
+r_planets = [r_mercury, r_venus, r_earth, r_mars,
+             r_jupiter, r_saturn, r_uranus, r_neptune]
+planet_tracks = [[], [], [], [], [], [], [], []]
+planet_actors = [mercury_actor, venus_actor, earth_actor, mars_actor,
+                 jupiter_actor, saturn_actor, uranus_actor, neptune_actor]
 
-        update_track(positions_venus, venus_track, venus_orbit_actor)
+for r_planet, planet_track in zip(r_planets, planet_tracks):
+    calculate_path(r_planet, planet_track, r_planet * 85)
 
-        update_track(positions_earth, earth_track, earth_orbit_actor)
+##############################################################################
+# This is for orbit visualization. We are using line actor for orbits.
+# After creating an actor we add it to the scene.
 
-        update_track(positions_mars, mars_track, mars_orbit_actor)
+for planet_track in planet_tracks:
+    orbit_actor = actor.line([planet_track], colors=(1, 1, 1), linewidth=0.1)
```

Comment:
this loop is not needed. `actor.line` can handle multiple line. I recommend you to make some tests. it will permit to have 1 actor instead of 8-9 line actors